### PR TITLE
Fixed Issue: Customer name in admin order management is "Guest"

### PR DESCRIPTION
**Pull request by mahesh-rajawat**
Fixed Issue of Place an order as an guest, the displayed customer name in the admin order management is "Guest". If the guest has created an account after placing the order, it is a link that points to the customer's account management. Invoice and shipping are also affected.

### Description (*)
If Order is placed by Guest user in admin Order view, Invoice view, Shipment view, displaying Customer
name as "Guest", issue was in the quote management, its only set email for the customer, Now i have it 
in this pull request.

### Fixed Issues (if relevant)
1. #18616: Customer name in adminhtml's order management is "Guest"

### Manual testing scenarios (*)
1. Created a new order as Guest for simple product, now Customer name is visible in admin order view.
2. Created a new order as Guest for virtual product as well, now Customer name is visible in admin order view.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by mahesh-rajawat** [comment]
#SQUASHTOBERFEST

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php [hunk 1]
```diff
@@ -222,14 +222,14 @@ public function execute()
...
                 );
-
-                if ($model->getIsUserDefined() === null) {
-                    $data['backend_type'] = $model->getBackendTypeByInput($data['frontend_input']);
-                }
             }
 
             $data += ['is_filterable' => 0, 'is_filterable_in_search' => 0];
 
+            if ($model->getIsUserDefined() === null || $model->getIsUserDefined() != 0) {
```
Comment (slavvka) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php:204]: Could you please tell what is the reason of changes in this file? How is it related to the original issue?

Comment (mahesh-rajawat) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php:204]: @slavvka you are right, I cloned my previous fixes and pushed in that so the previous files are also reflecting..... i will update it...

**Comment by mahesh-rajawat** [comment]
@slavvka update the other files which is not relevant to this issue.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php (hunk 1 above)
Comment (slavvka) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php:204]: @maheshWebkul721 Thank you! And please revert all empty lines that you removed.

Comment (slavvka) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php:204]: @maheshWebkul721 I would suggest you recreated your fix from the scratch and force-pushed it to ensure you don't have not needed changes.

**Review by slavvka — CHANGES_REQUESTED** [review]

**Comment by mahesh-rajawat** [comment]
@slavvka here i reference PR for this: #18928

**Pull request closed without merging**